### ISSUE TITLE
files 6.2.2 and 7.0.0 fail to compile on Fedora 40+

### ISSUE THREAD

**Issue by decathorpe** [elementary/files#2508]
When compiling files 7.0.0 on Fedora Rawhide, I get these errors in the build log:

```
(...)

../src/View/Sidebar/BookmarkListBox.vala: In function ‘sidebar_bookmark_list_box_constructor’:
../src/View/Sidebar/BookmarkListBox.vala:35:2: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
   35 |         list_box = new Gtk.ListBox () {
      |  ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     
      |  sidebar_sidebar_list_interface_get_list_box
../src/View/Sidebar/BookmarkListBox.vala: In function ‘_vala_sidebar_bookmark_list_box_set_property’:
../src/View/Sidebar/BookmarkListBox.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.BookmarkListBox : Gtk.Box, Sidebar.SidebarListInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_list_interface_get_sidebar

(...)

../src/View/Sidebar/BookmarkRow.vala:120:2: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_id’; did you mean ‘sidebar_sidebar_item_interface_get_id’? [-Wimplicit-function-declaration]
  120 |         id = SidebarItemInterface.get_next_item_id ();
      |  ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |  sidebar_sidebar_item_interface_get_id
../src/View/Sidebar/BookmarkRow.vala: In function ‘_vala_sidebar_bookmark_row_set_property’:
../src/View/Sidebar/BookmarkRow.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_list’; did you mean ‘sidebar_sidebar_item_interface_get_list’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_item_interface_get_list
../src/View/Sidebar/BookmarkRow.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_pinned’; did you mean ‘sidebar_sidebar_item_interface_get_pinned’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_item_interface_get_pinned
../src/View/Sidebar/BookmarkRow.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_item_interface_set_permanent’; did you mean ‘sidebar_sidebar_item_interface_get_permanent’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.BookmarkRow : Gtk.ListBoxRow, SidebarItemInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_item_interface_get_permanent

(...)

../src/View/Sidebar/DeviceListBox.vala:34:2: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
   34 |         list_box = new Gtk.ListBox () {
      |  ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     
      |  sidebar_sidebar_list_interface_get_list_box
../src/View/Sidebar/DeviceListBox.vala: In function ‘_vala_sidebar_device_list_box_set_property’:
../src/View/Sidebar/DeviceListBox.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.DeviceListBox : Gtk.Box, Sidebar.SidebarListInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_list_interface_get_sidebar

(...)

../src/View/Sidebar/NetworkListBox.vala: In function ‘sidebar_network_list_box_constructor’:
../src/View/Sidebar/NetworkListBox.vala:32:2: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_list_box’; did you mean ‘sidebar_sidebar_list_interface_get_list_box’? [-Wimplicit-function-declaration]
   32 |         list_box = new Gtk.ListBox () {
      |  ^      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~     
      |  sidebar_sidebar_list_interface_get_list_box
../src/View/Sidebar/NetworkListBox.vala: In function ‘_vala_sidebar_network_list_box_set_property’:
../src/View/Sidebar/NetworkListBox.vala:23:3: error: implicit declaration of function ‘sidebar_sidebar_list_interface_set_sidebar’; did you mean ‘sidebar_sidebar_list_interface_get_sidebar’? [-Wimplicit-function-declaration]
   23 | public class Sidebar.NetworkListBox : Gtk.Box, Sidebar.SidebarListInterface {
      |   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |   sidebar_sidebar_list_interface_get_sidebar

(...)
```

The version of vala is the same across all releases of Fedora (0.56.17), so it shouldn't be caused by a Vala upgrade ...

**Comment by ryonakano** [comment]
Duplicate of #2439 and fixed by #2459. This should be fixed in the next release of Files

**Comment by decathorpe** [comment]
Whoops, sorry about that. Thank you for the quick response!
